src/rewrite.rs

```diff
@@ -128,7 +166,34 @@ where
 {
     /// Search one eclass, returning None if no matches can be found.
     /// This should not return a SearchMatches with no substs.
-    fn search_eclass(&self, egraph: &EGraph<L, N>, eclass: Id) -> Option<SearchMatches<L>>;
+    ///
+    /// Implementation of [`Searcher`] should implement one of
```

Comment:
Hmm, I appreciate the effort to not break the API, but this could be a little confusing because the "default" behavior is an infinite loop! I think we should require `search_with_limit`.